cmd/skupper/skupper_kube_token.go

```diff
@@ -92,6 +108,97 @@ func (s *SkupperKubeToken) createFromTemplate(cmd *cobra.Command, args []string)
+		if err != nil {
+			return fmt.Errorf("failed getting kube secret: %s %w", TokenStatusOpts.Name, err)
+		}
+
+		fmt.Println("  namespace: ", cli.GetNamespace())
+		if claims, ok := secret.ObjectMeta.Annotations[types.ClaimsMade]; ok {
+			fmt.Println("  claims made: ", claims)
+		} else {
+			fmt.Println("  claims made: ", 0)
+		}
+		fmt.Println("  claims remaining: ", secret.ObjectMeta.Annotations[types.ClaimsRemaining])
+		if expirationStr, ok := secret.ObjectMeta.Annotations[types.ClaimExpiration]; ok {
+			expiration, timeErr := time.Parse(time.RFC3339, expirationStr)
+			if timeErr == nil {
+				if expiration.Before(time.Now()) {
+					fmt.Printf("  Warning: token has expired, expiration time: %s\n", expiration)
+				} else {
+					fmt.Printf("  expires: %s\n", expiration)
+				}
+			}
+		}
+	} else {
+		siteSecret, err := cli.KubeClient.CoreV1().Secrets(cli.GetNamespace()).Get(context.TODO(), types.SiteServerSecret, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed getting kube secret: %s %w", types.SiteServerSecret, err)
+		}
+
+		// Check that the ca.crt from the token matches the ca.crt from skupper-site-sevice
+		if strings.Compare(string(TokenStatusOpts.Secret.Data[types.ClaimCaCertDataKey]), string(siteSecret.Data[types.ClaimCaCertDataKey])) == 0 {
+			fmt.Printf("  CA: Token matches %s\n", types.SiteServerSecret)
+		} else {
+			fmt.Println("  CA: does not match")
+		}
+
+		cert, err := certs.DecodeCertificate(siteSecret.Data["tls.crt"])
+		if err != nil {
+			return fmt.Errorf("failed to DecodeCertificate for %s %w", types.SiteServerSecret, err)
+		}
+
+		cert2, err := certs.DecodeCertificate(TokenStatusOpts.Secret.Data["tls.crt"])
+		if err != nil {
+			return fmt.Errorf("failed to DecodeCertificate for %s %w", TokenStatusOpts.Name, err)
+		}
+
+		// check that the hostname/IP is in the SANs list
+		if secretHostname, ok := TokenStatusOpts.Secret.ObjectMeta.Annotations["inter-router-host"]; ok {
+			hostFound := false
+			for _, hostname := range cert.DNSNames {
+				if secretHostname == hostname {
+					hostFound = true
+					fmt.Printf("  hostname: %s found in SANs list\n", secretHostname)
+					break
+				}
+			}
+			if !hostFound {
+				fmt.Printf("  hostname: %s not found in SANs list\n", secretHostname)
+			}
+		}
+
+		// check that skupper is the issuer of both secrets
+		if cert.Issuer.CommonName == cert2.Issuer.CommonName {
+			fmt.Printf("  Issuer: %s for both\n", cert.Issuer.CommonName)
+		}
+
+		fmt.Println()
+		fmt.Println("Skupper-site-server Secret details:")
+		fmt.Println("  AuthorityKeyId:", cert.AuthorityKeyId)
+		fmt.Println("  DNS:", cert.DNSNames)
+		fmt.Println("  IP:", cert.IPAddresses)
+		fmt.Println("  Issuer:", cert.Issuer)
+		fmt.Println("  Valid not before:", cert.NotBefore)
+		fmt.Println("  Valid not after:", cert.NotAfter)
+
+		fmt.Println()
+		fmt.Println("Token Secret: details")
```

Comment:
The colon here is in a different position that for the skupper-site-secret details, which is trivial but looks a bit odd.

Reply:
fixed